feature-libs/checkout/base/components/services/express-checkout.service.ts

```diff
@@ -0,0 +1,169 @@
+import { Injectable } from '@angular/core';
+import {
+  CheckoutDeliveryAddressFacade,
+  CheckoutDeliveryModesFacade,
+  CheckoutPaymentFacade,
+} from '@spartacus/checkout/base/root';
+import {
+  FeatureConfigService,
+  UserAddressService,
+  UserPaymentService,
+} from '@spartacus/core';
+import { combineLatest, Observable, of } from 'rxjs';
+import {
+  catchError,
+  debounceTime,
+  filter,
+  map,
+  switchMap,
+  take,
+  tap,
+} from 'rxjs/operators';
+import { CheckoutConfigService } from '../services/checkout-config.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ExpressCheckoutService {
+  private shippingAddressSet$: Observable<boolean>;
+  private deliveryModeSet$: Observable<boolean>;
+  private paymentMethodSet$: Observable<boolean>;
+
+  constructor(
+    protected userAddressService: UserAddressService,
```

Comment:
Can you remind me

I know that express checkout wasn't entirely updated to c&q due to the fact that we have 2 dependencies that are non relevant to checkout (UserAddressService and UserPaymentService)

The conclusion of this file was that we leave the ngrx as it is and just update the checkout dependencies and logic accordingly right?

Reply:
If you are referring to the usage of loading and success flags, then yes - as soon as we polish those APIs, we can improve the express-checkout as well.